test-data/unit/check-protocols.test

```diff
@@ -2219,6 +2219,17 @@ if issubclass(cls, P):
     reveal_type(cls)  # N: Revealed type is 'Type[__main__.C]'
 else:
     reveal_type(cls)  # N: Revealed type is 'Type[__main__.E]'
+
+@runtime_checkable
+class POverload(Protocol):
+    @overload
+    def meth(self, a: int) -> float: ...
+    @overload
+    def meth(self, a: str) -> Sequence[float]: ...
+    def meth(self, a):
+        pass
+
+reveal_type(issubclass(int, POverload))  # N: Revealed type is 'builtins.bool'
```

Comment:
Could you also add a concrete class (e.g., `COverload`) that has a suitable overloaded method, and test that `issubclass` can be used to narrow down from `Type[Union[COverload, E]]`, similar to above around line 2218?

Reply:
So while trying the add the test you proposed I seem to have stumbled across another bug.

The `if` block successfully narrows down the `Union` to `Type[COverload]` but the `else` block
on the other hand fails to do the same with `E`, revealing the type as the original `Union`.

Would you happen to know where in mypy the type inference of `if` / `else` statements is handled?
I imagine that change similar to `non_method_protocol_members()` will have to be made there,
but I'm not quite sure where to look.

Examples
---------
``` python
class COverload:
    x: str
    @overload
    def meth(self, a: int) -> float: ...
    @overload
    def meth(self, a: str) -> Sequence[float]: ...
    def meth(self, a):
        pass

cls_overload: Type[Union[COverload, E]]
if issubclass(cls_overload, POverload):
    reveal_type(cls_overload)  # N: Revealed type is 'Type[__main__.COverload]'
else:
    reveal_type(cls_overload)  # N: Revealed type is 'Type[__main__.E]'
```

The test output:
```
Expected:
  ...
  main:45: note: Revealed type is 'Type[__main__.COverload]'
  main:47: note: Revealed type is 'Type[__main__.E]' (diff)
Actual:
  ...
  main:45: note: Revealed type is 'Type[__main__.COverload]'
  main:47: note: Revealed type is 'Union[Type[__main__.COverload], Type[__main__.E]]' (diff)
```

Reply:
Could you please open a new issue about it?

Reply:
> Could you please open a new issue about it?

I just created an issue at https://github.com/python/mypy/issues/12228.